ntbs-service-unit-tests/Services/NtbsSearchBuilderTest.cs

```diff
@@ -53,13 +53,13 @@ public NotificationSearchBuilderTest()
                 },
             }).AsQueryable();
 
-            builder = new NotificationSearchBuilder(notifications);
+            builder = new NtbsSearchBuilder(notifications);
         }
 
         [Fact]
         public void SearchById_ReturnsMatchOnNotificationId()
         {
-            var result = builder.FilterById("1").GetResult().ToList();
+            var result = ((INtbsSearchBuilder)builder.FilterById("1")).GetResult().ToList();

```

Comment:
builder.FilterById("1") was casting(converting?) the builder to the parent which FilterById is defined on (SearchBuilder) which does not have the GetResult method and the compiler was complaining so I needed to explicitly cast this to the correct NtbsSearchBuilder

Reply:
yup, quite reasonable - especially that in code we'll rely on knowing which one it is, too